### DISCUSSION TITLE
请教一下这个错误是什么原因，我的配置有什么问题？

### DISCUSSION THREAD

**Discussion by kursk-ye** [XTLS/Xray-core#528] [Q&A]
我作用V2RAY仅仅作DNS的解析和转发，客户端是Openwrt，操作系统和配置如下

客户端

操作系统版本
openwrt_release  openwrt_version
root@OpenWrt:/tmp/tmp# cat /etc/openwrt_release
DISTRIB_ID='OpenWrt'
DISTRIB_RELEASE='18.06.5'
DISTRIB_REVISION='r7897-9d401013fc'
DISTRIB_TARGET='mvebu/cortexa9'
DISTRIB_ARCH='arm_cortex-a9_vfpv3'
DISTRIB_DESCRIPTION='OpenWrt 18.06.5 r7897-9d401013fc'
DISTRIB_TAINTS=''

root@OpenWrt:/tmp/tmp# xray --version
Xray 1.4.2 (Xray, Penetrates Everything.) Custom (go1.16 linux/arm)
A unified platform for anti-censorship.

json配置文件
```
{
  "log": {
    "loglevel": "debug"
  },
  "inbound": {
    "sniffing": {
      "enabled": true,
      "destOverride": [
        "http",
        "tls"
      ]
    },
    "port": 5301,
    "protocol": "dokodemo-door",
    "settings": {
      "network": "tcp,udp",
      "followRedirect": true
    }
  },
  "outbounds":[
    {
      "tag": "dns",
      "protocol": "vless",
      "settings": {
        "vnext": [
          {
            "address": "123.xyz",
            "port": 443,
            "users": [
              {
                "id": "11e83032-f583-e090-051e-**",
                "alterId": 0,
                "email": "123.xyz_VLESS_XTLS/TLS-direct_TCP",
                "security": "auto",
                "encryption": "none",
                "flow": ""
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "tcp",
        "security": "tls",
        "tlsSettings": {
          "allowInsecure": true,
          "serverName": "123.xyz"
        }
      },
      "mux": {
        "enabled": false,
        "concurrency": -1
      }
    }
  ]
}
```

服务端

操作系统
root@Net2:~# cat /etc/issue
Ubuntu 20.04 LTS \n \l

v2ray版本
root@Net2:~# /etc/v2ray-agent/v2ray/v2ray --version
V2Ray 4.32.1 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.15.4 linux/amd64)
A unified platform for anti-censorship.

服务端配置

root@Net2:/etc/v2ray-agent/v2ray/conf# cat 12_dns_forward.json
```
{
  "inbounds": [
      {
          "port": 443,
          "protocol": "dokodemo-door",
          "tag": "dns",
          "settings": {
            "clients": [
             {
                "id": "11e83032-f583-e090-051e-***",
                "add":"123.xyz",
                "flow":"",
                "email": "123.xyz_VLESS_XTLS/TLS-direct_TCP"
              }
            ],
            "decryption": "none",
            "fallbacks": [
                {"dest":31296,"xver":0},{"path":"/zmzvws","dest":31297,"xver":1},{"path":"/zmzvtcp","dest":31298,"xver":1},{"path":"/zmzvvws","dest":31299,"xver":1}
            ]
          },
          "streamSettings": {
            "network": "tcp",
            "security": "tls",
            "tlsSettings": {
              "alpn": [
                "http/1.1"
              ],
              "certificates": [
                {
                  "certificateFile": "/etc/v2ray-agent/tls/123.xyz.crt",
                  "keyFile": "/etc/v2ray-agent/tls/123.xyz.key"
                }
              ]
            }
          }
      }
  ],
  "outbounds": [
      {
          "protocol": "freedom",
          "tag": "dns"
      }
  ]
}
```

但是启动后报如下错误

2021/04/11 10:45:37 127.0.0.1:46103 accepted udp:127.0.0.1:5301 [dns]
2021/04/11 10:45:38 [Info] [3287941024] proxy/vless/outbound: tunneling request to udp:127.0.0.1:5301 via tcp:kurskonworld.xyz:443
2021/04/11 10:45:53 [Info] [3287941024] app/proxyman/outbound: failed to process outbound traffic > proxy/vless/outbound: connection ends > proxy/vless/outbound: failed to decode response header > proxy/vless/encoding: failed to read response version > EOF
2021/04/11 10:45:53 [Info] [3287941024] app/proxyman/inbound: connection ends > proxy/dokodemo: connection ends > proxy/dokodemo: failed to transport response > io: read/write on closed pipe

请问问题可能是什么，是我哪里配置不正确，我应该如何修改配置？

**Comment by xqzr** [comment]
服务端配置的 `"protocol": "dokodemo-door"` 改为 `"protocol": "vless"`
